Library/Homebrew/utils/tty.rb

```diff
@@ -66,6 +75,16 @@ def reset_escape_sequence!
     end
   end
 
+  SPECIAL_CODES.each do |name, code|
+    define_singleton_method(name) do
+      if $stdout.tty?
```

Comment:
I'm not sure of the best way to update this but it needs to be modified in some way. Right now, the progress bar isn't cleared at the end of the run when `$stdout` is redirected to a file (e.g., `brew livecheck --json > test.json`).

Reply:
Despite `stderr.tty?` being true, I don't see coloured output (the `Running checks` line), is there something I'm missing?

Reply:
My tentative solution for this was to replace this line with `if $stdout.tty? || $stderr.tty?`. Fixing the color issue involves a similar change in `#color?`, where we could change `$stdout.tty?` to `$stdout.tty? || $stderr.tty?`.

That said, I'm not sure if this is the most appropriate way to handle this. It works for our particular purposes but it would be worth making sure that this won't negatively impact some other usage of `utils/tty` in Homebrew.

Reply:
Got it 👍🏼 I'll dig a little deeper and see what's going on. Thanks Sam.

Reply:
I went through the usages after identifying them using `grep -R`, and I don't think there's going to be any untoward impact. So I've gone ahead and made this change.